client/src/panels/transfer_payments/tp_by_region.js

```diff
@@ -0,0 +1,119 @@
+import text from './tp_by_region.yaml';
+import {
+  formats,
+  run_template,
+  declare_panel,
+  years,
+  businessConstants,
+  create_text_maker_component,
+  StdPanel,
+  Col,
+  declarative_charts,
+} from "../shared";
+import { ExtendedCanada } from '../../charts/extended_canada.js';
+
+const { std_years } = years;
+const context_years = std_years;
+const formatter = formats["compact2_raw"];
+const includeNcr = false;
+
+const { text_maker } = create_text_maker_component(text);
+const { provinces } = businessConstants;
+const { A11YTable } = declarative_charts;
+
+const calculate_common = (years_by_province) => {
+  const max = d3.max(d3.values(_.last(years_by_province)));
+  const color_scale = d3.scaleLinear()
+    .domain([0,max])
+    .range([0.2,1]);
+  return {
+    years_by_province,
+    color_scale,
+    context_years,
+    formatter,
+    includeNcr,
+  };
+};
+
+const calculate_funcs_by_level = {
+  gov: function(){
+    const { orgTransferPaymentsRegion } = this.tables;
+    return calculate_common( std_years.map( year => orgTransferPaymentsRegion.prov_code(year, false) ) );
+  },
+  dept: function(subject){
+    const { orgTransferPaymentsRegion } = this.tables;
+    return calculate_common( std_years.map( year => orgTransferPaymentsRegion.prov_code(year, subject.unique_id) ) );
+  },
+};
+
+const prepare_data_for_a11y_table = (years_by_province) => {
+  const total_tp = _.chain(years_by_province)
+    .map( year_by_province => d3.sum( _.values(year_by_province) ) )
+    .reduce( (sum, value) => sum + value, 0)
+    .value();
+  
+  return _.chain(provinces)
+    .map((province, prov_code) => {
+      if(prov_code!="onlessncr" && prov_code!="qclessncr" && prov_code!="ncr"){
+        const yearly_tp = _.map(years_by_province, (year_by_province) => year_by_province[prov_code]);
+        const formatted_yearly_tp = _.map(years_by_province, (year_by_province) => formats["compact2_written_raw"](year_by_province[prov_code]));
+        const five_year_avg_share = d3.sum(yearly_tp)/total_tp;
+        const formatted_avg_share = formats["percentage1_raw"](five_year_avg_share);
```

Comment:
All the average share stuff was only ever done in the people tables. I figure it makes sense there because the demographics are fairly stable, and people are generally curious about how demographics are split up by percentage. I think with money, people prefer to get hard values (even if what they really want probably is the distribution in this case), and fluctuations tend to be more common/meaningful. Big dollar values are just exciting a lot more exciting than percents!

Just my opinion, nothing to back these claims up really.

Reply:
okay yeah that makes sense so are you suggesting to remove average share column? (As you know this is for A11y table column!) Or something like how much money increased/decreased over 5 years? Just asking to be clear!

Reply:
I say drop it, yeah. Thanks!